TFNetwork.py

```diff
@@ -388,6 +388,12 @@ def construct_layer(self, net_dict, name, get_layer=None, add_layer=None, check_
         layer_desc = {"class": "source", "from": []}
       elif name.startswith("data:"):
         layer_desc = {"class": "source", "data_key": name[len("data:"):], "from": []}
+      elif '/' in name:
+        # it may be a hierarchical path to a sub-layer, which should have been found by get_layer()
+        # but maybe it's not constructed yet, so try constructing the root layer
+        self.construct_layer(net_dict, name.split('/')[0], get_layer, add_layer, check_existing)
+        # constructing the root layer should have constructed all its children
+        return self.get_layer(name)  # ...so try again now
```

Comment:
It should not call `self.construct_layer` in all cases, but rather `get_layer` (the param passed to this function).
Thus, move that code `if not get_layer: ...`, which is below, above this, and then call `get_layer` here.